[CI] KeyRotationIT testEachInstallOrRotationAddsExactlyOneNewKey failing

**Build Scans:**
- [elasticsearch-intake #42698 / part2](https://gradle-enterprise.elastic.co/s/j34srxzot37ia)
- [elasticsearch-intake #42608 / part2](https://gradle-enterprise.elastic.co/s/ekbzhxnnzy5sg)
- [elasticsearch-pull-request #158985 / part-2](https://gradle-enterprise.elastic.co/s/vhtkuanzmomcw)
- [elasticsearch-pull-request #158985 / part-2](https://gradle-enterprise.elastic.co/s/47ux7df26r5qm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:encryption:internalClusterTest" --tests "org.elasticsearch.xpack.encryption.KeyRotationIT.testEachInstallOrRotationAddsExactlyOneNewKey" -Dtests.seed=3542A4D7BE959403 -Dtests.locale=nds-Latn-DE -Dtests.timezone=Asia/Chongqing -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.encryption.KeyRotationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEachInstallOrRotationAddsExactlyOneNewKey'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[PendingClusterTask[insertOrder=95, priority=NORMAL, source=begin-project-encryption-key-rotation, timeInQueue=10019, executing=true], PendingClusterTask[insertOrder=96, priority=NORMAL, source=retire-project-encryption-keys, timeInQueue=10019, executing=true], PendingClusterTask[insertOrder=97, priority=NORMAL, source=retire-project-encryption-keys, timeInQueue=9014, executing=false], PendingClusterTask[insertOrder=98, priority=NORMAL, source=retire-project-encryption-keys, timeInQueue=8013, executing=false], PendingClusterTask[insertOrder=99, priority=NORMAL, source=retire-project-encryption-keys, timeInQueue=7013, executing=false], PendingClusterTask[insertOrder=100, priority=NORMAL, source=retire-project-encryption-keys, timeInQueue=6012, executing=false], PendingClusterTask[insertOrder=95, priority=NORMAL, source=begin-project-encryption-key-rotation, timeInQueue=10019, executing=true], PendingClusterTask[insertOr
[truncated]
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testEachInstallOrRotationAddsExactlyOneNewKey
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 4 failures in test testEachInstallOrRotationAddsExactlyOneNewKey (0.4% fail rate in 940 executions)
- [main] 2 failures in step part2 (1.6% fail rate in 122 executions)
- [main] 2 failures in step part-2 (0.3% fail rate in 685 executions)
- [main] 2 failures in pipeline elasticsearch-intake (1.6% fail rate in 122 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -826,9 +826,6 @@ tests:
 - class: org.elasticsearch.search.basic.SearchWithRandomDisconnectsIT
   method: testSearchWithRandomDisconnects
   issue: https://github.com/elastic/elasticsearch/issues/152734
-- class: org.elasticsearch.xpack.encryption.KeyRotationIT
-  method: testEachInstallOrRotationAddsExactlyOneNewKey
-  issue: https://github.com/elastic/elasticsearch/issues/152776
 - class: org.elasticsearch.transport.RemoteClusterClientTests
   method: testGetConnectionFailureCompletesOnResponseExecutor
   issue: https://github.com/elastic/elasticsearch/issues/152779

